### PR TITLE
use libevm: params, core/vm, eth/tracers/* + some of core/types

### PR DESCRIPTION
**Pull request by darioush**
## Why this should be merged
Applies same changes as https://github.com/ava-labs/subnet-evm/issues/662

## How this works
Same as 662.

## How this was tested
CI

## Need to be documented?
N/A

## Need to update RELEASES.md?
N/A (we can update it with transition to libevm)

### CODE REVIEW

File: scripts/geth-allowed-packages.txt
Comment (ARR4N) [inline scripts/geth-allowed-packages.txt:1]: 😍

**Review by ARR4N — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -152,3 +153,7 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
+
+replace github.com/ethereum/go-ethereum => github.com/ava-labs/go-ethereum v0.0.0-20241007222654-0752a11d4aee
```
[the remark this answers is not in the document]
Comment (darioush) [reply go.mod:157]: yes

File: params/hooks_libevm.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		if !allowListRole.IsEnabled() {
+			gas = 0
+			return gas, fmt.Errorf("tx.origin %s is not authorized to deploy a contract", ac.Origin)
+		}
+	}
+
+	return gas, nil
+}
+
+func (r RulesExtra) CanExecuteTransaction(_ common.Address, _ *common.Address, _ libevm.StateReader) error {
```
Comment (ceyonur) [inline params/hooks_libevm.go:42]: should we use the txallowlist here?

Comment (darioush) [reply params/hooks_libevm.go:42]: Added a TODO about this

**Review by ceyonur — APPROVED** [review]

**Pull request merged**